backend_new/src/services/application-csv-export.service.ts

```diff
@@ -0,0 +1,655 @@
+import fs, { createReadStream } from 'fs';
+import path, { join } from 'path';
+import { Injectable, StreamableFile } from '@nestjs/common';
+import { Request as ExpressRequest } from 'express';
+import { view } from './application.service';
+import { PrismaService } from './prisma.service';
+import { MultiselectQuestionService } from './multiselect-question.service';
+import { ApplicationCsvQueryParams } from '../dtos/applications/application-csv-query-params.dto';
+import { SuccessDTO } from '../dtos/shared/success.dto';
+import { UnitType } from '../dtos/unit-types/unit-type.dto';
+import { Address } from '../dtos/addresses/address.dto';
+import { ApplicationMultiselectQuestion } from '../dtos/applications/application-multiselect-question.dto';
+import MultiselectQuestion from '../dtos/multiselect-questions/multiselect-question.dto';
+import { ApplicationFlaggedSet } from '../dtos/application-flagged-sets/application-flagged-set.dto';
+
+view.csv = {
+  ...view.details,
+  applicationFlaggedSet: true,
+  listings: false,
+};
+
+type CsvHeader = {
+  path: string;
+  label: string;
+  format?: (val: unknown) => unknown;
+};
+
+const typeMap = {
+  SRO: 'SRO',
+  studio: 'Studio',
+  oneBdrm: 'One Bedroom',
+  twoBdrm: 'Two Bedroom',
+  threeBdrm: 'Three Bedroom',
+  fourBdrm: 'Four+ Bedroom',
+};
+
+@Injectable()
+export class ApplicationCsvExporterService {
+  constructor(
+    private prisma: PrismaService,
+    private multiselectQuestionService: MultiselectQuestionService,
+  ) {}
+  /**
+   *
+   * @param queryParams
+   * @param req
+   * @returns a promise containing a streamable file
+   */
+  async export(
+    queryParams: ApplicationCsvQueryParams,
+    req: ExpressRequest,
+  ): Promise<StreamableFile> {
+    await this.authorizeCSVExport(req.user, queryParams.listingId);
+    const filename = join(
+      process.cwd(),
+      `src/temp/listing-${
+        queryParams.listingId
+      }-applications-${new Date().getTime()}.csv`,
+    );
+    await this.createCsv(filename, queryParams);
+    const file = createReadStream(filename);
+    file.on('end', () => {
+      fs.unlink(filename, () => {
+        console.log(`deleted ${filename}`);
```

Comment:
was this console log added here for testing purposes?

Reply:
yes, removed and added error handling